### ISSUE TITLE
Setting requireBranch property to a regular expressions

### ISSUE THREAD

**Issue by grigorigoldman** [researchgate/gradle-release#118]
We do most of our releases from integration branches which are normally named something like this: integration_1_0, integration_2_1, etc.

It would be great if I could set the requireBranch in my script to a regular expression, something like this `/integration_.*/`. Unfortunately, the check in the `GitAdapter.init` method is doing a straight string comparison.

Would it be possible, to change this check from:

```
branch != extension.git.requireBranch
```

to something like this:

```
!(branch ==~  extension.git.requireBranch)
```

If you think this is a good idea, I'm happy to submit a PR.

**Comment by danez** [comment]
That sounds reasonable. (y)

**Comment by danez** [comment]
Fixed with #119
